spa/spa.go

```diff
@@ -0,0 +1,48 @@
+package spa
+
+import (
+	"embed"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+
+	"github.com/NYTimes/gziphandler"
+)
+
+// Init creates a single page application manager.
+func Init(build embed.FS, dir string, index string) (*SPA, error) {
+	fsys, err := fs.Sub(build, dir)
+	if err != nil {
+		panic(fmt.Errorf("couldn't create sub filesystem: %w", err))
```

Comment:
why panic here?

Reply:
Yes, we can avoid this and let users of the package decide what they want to do.

Reply:
Fixed.